src/test/java/com/tenx/ms/retail/rest/OrderControllerTest.java

```diff
@@ -0,0 +1,169 @@
+package com.tenx.ms.retail.rest;
+
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.tenx.ms.commons.config.Profiles;
+import com.tenx.ms.commons.rest.RestConstants;
+import com.tenx.ms.commons.tests.AbstractIntegrationTest;
+import com.tenx.ms.retail.RetailServiceApp;
+import com.tenx.ms.retail.rest.dto.Order;
+import com.tenx.ms.retail.rest.dto.OrderItem;
+import com.tenx.ms.retail.rest.dto.OrderResult;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.SpringApplicationConfiguration;
+import org.springframework.boot.test.TestRestTemplate;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.web.client.RestTemplate;
+
+import java.io.IOException;
+import java.util.ArrayList;
+
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.fail;
+
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringApplicationConfiguration(classes = RetailServiceApp.class)
+@ActiveProfiles(Profiles.TEST_NOAUTH)
+public class OrderControllerTest extends AbstractIntegrationTest {
+
+    private final RestTemplate template = new TestRestTemplate();
+    private final String API_VERSION = RestConstants.VERSION_ONE;
+    private final String REQUEST_URI = "%s" + API_VERSION + "/orders/";
+    private static final Long INVALID_PRODUCT_ID = 99999999L;
+
+    @Autowired
+    private ObjectMapper mapper;
+
+    @Test
+    public void testCreateNewOrder() {
+
+        Long storeId = 1L;
+        Order order = new Order();
+
+        order.setFirstName("Petter");
+        order.setLastName("Parker");
+        order.setStoreId(1L);
+        order.setEmail("pparker@marvel.com");
+        order.setPhone("1234567890");
+
+        ArrayList<OrderItem> items = new ArrayList<>();
```

Comment:
You should always be assigning to the interface in Java. Just like how you use interfaces within Method signatures.

`List<OrderItem> items = new ArrayList<>();`